jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/AdvanceRuntimeDataServiceImplTest.java

```diff
@@ -682,4 +683,165 @@ public void testParameterWithDash() {
+    }
+
+    private void assertQueryProcessByVariablesAndTaskSorted(List<QueryParam> attributes, List<QueryParam> variables) {
+        List<ProcessInstanceWithVarsDesc> actualDataList = advanceVariableDataService.queryProcessByVariablesAndTask(attributes, variables, emptyList(), emptyList(), new QueryContext());
+
+        List<Long> expectedProcessInstanceIdsList = actualDataList
+                .stream()
+                .map(ProcessInstanceWithVarsDesc::getId)
+                .collect(Collectors.toList());
+        List<String> expectedCorrelationKeysList = actualDataList
+                .stream()
+                .map(ProcessInstanceWithVarsDesc::getCorrelationKey)
+                .collect(Collectors.toList());
+        assertThat(expectedProcessInstanceIdsList.size(), is(6));
+        assertThat(expectedCorrelationKeysList.size(), is(6));
+
+        // Order by processInstanceId ascending
+        queryContext.setOrderBy("processInstanceId");
+        queryContext.setAscending(true);
+        actualDataList = advanceVariableDataService.queryProcessByVariablesAndTask(attributes, variables, emptyList(), emptyList(), queryContext);
+        assertSortedListByColumn(true, expectedProcessInstanceIdsList, actualDataList.stream().map(ProcessInstanceWithVarsDesc::getId).collect(Collectors.toList()));
+
+        // Order by processInstanceId descending
+        queryContext.setAscending(false);
+        actualDataList = advanceVariableDataService.queryProcessByVariablesAndTask(attributes, variables, emptyList(), emptyList(), queryContext);
+        assertSortedListByColumn(false, expectedProcessInstanceIdsList, actualDataList.stream().map(ProcessInstanceWithVarsDesc::getId).collect(Collectors.toList()));
+
+        // Order by correlationKey ascending
+        queryContext.setOrderBy("correlationKey");
+        queryContext.setAscending(true);
+        actualDataList = advanceVariableDataService.queryProcessByVariablesAndTask(attributes, variables, emptyList(), emptyList(), queryContext);
+        assertSortedListByColumn(true, expectedCorrelationKeysList, actualDataList.stream().map(ProcessInstanceWithVarsDesc::getCorrelationKey).collect(Collectors.toList()));
+
+        // Order by correlationKey descending
+        queryContext.setAscending(false);
+        actualDataList = advanceVariableDataService.queryProcessByVariablesAndTask(attributes, variables, emptyList(), emptyList(), queryContext);
+        assertSortedListByColumn(false, expectedCorrelationKeysList, actualDataList.stream().map(ProcessInstanceWithVarsDesc::getCorrelationKey).collect(Collectors.toList()));
+    }
+
+    private <T extends Comparable<? super T>> void assertSortedListByColumn(boolean ascending, List<T> expectedSortedList, List<T> actualSortedList){
+        int count = queryContext.getCount() > 0 ? queryContext.getCount() : expectedSortedList.size();
+        if (ascending) {
```

Comment:
you can write a shorter version
Collections.sort(list, ascending ? null : Collections.reverseOrder());

Reply:
yes, that's better. Thanks